src/backend/distributed/executor/multi_executor.c

```diff
@@ -232,6 +232,12 @@ CitusExecutorRun(QueryDesc *queryDesc,
 			 * transactions.
 			 */
 			CitusTableCacheFlushInvalidatedEntries();
+
+			/*
+			 * Now that the function call is completed, release the saved
+			 * (if any) distribution argument value of the function.
```

Comment:
We get here at the end of any query, so this comment should be a little more precise about which case this covers.

Reply:
Added more detail to the comment